1.Exploratory_Data_Analysis/scripts/5.anova.py

```diff
@@ -0,0 +1,385 @@
+    posthoc = pairwise_tukeyhsd(
+        combined_df[i],
+        combined_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"],
+        alpha=0.001,
+    )
+    # print(posthoc)
+    lst.append([posthoc, i])
+
+
+# In[8]:
+
+
+# add all tukey test results to a dataframe from the list of tukey test results
+tukey_df = pd.DataFrame()
+for i in lst:
+    # create data frame from tukey test results with feature name as column name and the results as the rows
+    j = pd.DataFrame(i[0]._results_table.data[1:])
+    j["features"] = np.repeat(i[1], len(j))
+    tukey_df = pd.concat([tukey_df, j], axis=0)
+
+    np.repeat(i[1], len(j))
+
+tukey_df.columns = [
+    "group1",
+    "group2",
+    "meandiff",
+    "lower",
+    "upper",
+    "p-adj",
+    "reject",
+    "features",
+]
+tukey_df.head(3)
+
+
+# In[9]:
+
+
+# make new column with the absolute value of the p-adj
+tukey_df["p-adj_abs"] = abs(tukey_df["p-adj"])
+# get all p-adj values that are less than 0.05
+tukey_df_sig = tukey_df[tukey_df["p-adj_abs"] < 0.01]
+# make new column that states if the relationship is positive or negative
+tukey_df_sig["pos_neg"] = np.where(tukey_df_sig["p-adj"] > 0, "positive", "negative")
+# order the features by p-adj value
+tukey_df_sig.head(2)
+
+
+# ## Venn Diagrams
+
+# #### Venn diagram prep
+
+# In[10]:
+
+
+# get all group1 rows that are DMSO and group2 rows that are LPS 100
+tukey_df_sig_trt_1v2 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment1}")
+    & (tukey_df_sig["group2"] == f"{treatment2}")
+]
+# get all group1 rows that are DMSO and group2 rows that are Thapsigargin 10
+tukey_df_sig_trt_1v3 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment1}")
+    & (tukey_df_sig["group2"] == f"{treatment3}")
+]
+# get all group1 rows that are LPS 100 and group2 rows that are Thapsigargin 10
+tukey_df_sig_trt_2v3 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment2}")
+    & (tukey_df_sig["group2"] == f"{treatment3}")
+]
+
+
+# In[11]:
+
+
+from matplotlib import rcParams
+
+rcParams.update({"figure.autolayout": True})
+
+
+# ### Venn Diagram 2 groups
+
+# In[12]:
+
+
+venn2(
+    [set(tukey_df_sig_trt_1v3["features"]), set(tukey_df_sig_trt_2v3["features"])],
+    set_labels=(f"{treatment1} vs {treatment2}", f"{treatment1} vs {treatment3}"),
+)
+plt.title("Number of Significant (p-adj < 0.01) Features \n per Organelle", size=24)
+save_path = pathlib.Path(
+    f"./Figures/anova_of_features/{treatment1}_vs_{treatment2}_vs_{treatment3}"
+)
+save_path.mkdir(parents=True, exist_ok=True)
+# plt.tight_layout()
+# plt.autoscale()
+plt.subplots_adjust(left=0.15)
+plt.savefig(
+    f"{save_path}/{treatment1}_vs_{treatment2}_number_sig_overlaping_features_per_organelle.png",
+    dpi=300,
+    bbox_inches="tight",
+)
+plt.show()
+
+
+# In[ ]:
+
+
+# In[13]:
+
+
+# create a venn diagram of the features that are significant in all conditions
+from matplotlib_venn import venn2, venn3_unweighted
```

Comment:
Same as above comment, recommended that all imports are at the beginning of the file.